packages/base/src/UI5Element.js

```diff
@@ -129,13 +143,16 @@ class UI5Element extends HTMLElement {
 	 * @private
 	 */
 	disconnectedCallback() {
-		if (!this.constructor._needsShadowDOM()) {
-			return;
+		if (this.constructor._needsShadowDOM()) {
+			this._stopObservingDOMChildren();
+			if (typeof this.onExitDOM === "function") {
+				this.onExitDOM();
+			}
 		}
 
-		this._stopObservingDOMChildren();
-		if (typeof this.onExitDOM === "function") {
-			this.onExitDOM();
+		if (typeof this.constructor.staticAreaTemplate === "function") {
+			const staticAreaItemToRemove = document.querySelector(`static-area-item-${this.currentStaticAreaPosition}`);
```

Comment:
This could become a method of the StaticArea class that will be moved to a separate module